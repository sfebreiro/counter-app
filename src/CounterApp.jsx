import { useState, Fragment } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

        const [ counter, setCounter ] = useState( value );

        const plusOne = () => {
            // console.log(event);
            setCounter ( counter + 1 );
        }

        const minusOne = () => setCounter ( counter - 1 );

        const resetCounter = () => setCounter ( value );
        
        return(
        
            <Fragment>
                <h1>CounterApp</h1>

                <h2> { counter } </h2>

                <button onClick={ plusOne }>
                    +1
                </button>

                <button onClick={ minusOne }>
                    -1
                </button>

                <button aria-label='btn-reset' onClick={ resetCounter }>
                    Reset
                </button>
            </Fragment>
        
    )
} 


CounterApp.protTypes = {
    value: PropTypes.number.isRequired
}


