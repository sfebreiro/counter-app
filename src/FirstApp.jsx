import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle, name}) => {

        return(
        <>
        {/* <code> {JSON.stringify(newMessage)} </code> */}
        <h1 data-testid="test-title">{title}</h1>
        <p>{subtitle}</p>
        <p>{subtitle}</p>
        <p>{name}</p>
        </>
    )
} 


FirstApp.protTypes = {
    title: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'No hay subt√≠tulo',
    name: 'Samu'
}
