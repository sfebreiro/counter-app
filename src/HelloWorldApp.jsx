// function App() {
//     return <h1>Hola React!!</h1>
// }

// export default App;

import React from 'react'

export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>
  )
}
