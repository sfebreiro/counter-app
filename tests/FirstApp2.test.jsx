import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en FirstApp', () => {  

    const title = 'Hola, tengo una sudadera nueva';
    const subtitle = 'No hay subtítulo';

    test('debe hacer match con el snapshot' , () => { 
        
        const {container} = render(<FirstApp title={title} /> ) 
        expect(container).toMatchSnapshot();
    
    });


    test('debe mostrar el mensaje: Hola, tengo una sudadera nueva' , () => { 
        
        render(<FirstApp title={title} /> ) 
        expect(screen.getByText(title)).toBeTruthy();
    
    });


    test('debe mostrar el título en un h1' , () => { 
        
        render(<FirstApp title={title} /> ) 
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title);
    
    });



    test('debe mostrar el subtítulo enviado por las props' , () => { 
        
        render(<FirstApp title={title} subtitle={subtitle}/> ) 
        expect(screen.getAllByText(subtitle).length).toBe(2);
    
    });

});