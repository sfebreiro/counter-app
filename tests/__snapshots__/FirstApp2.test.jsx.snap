// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, tengo una sudadera nueva
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Samu
  </p>
</div>
`;
